util/route.libsonnet

```diff
@@ -17,6 +17,31 @@ local d = import 'github.com/jsonnet-libs/docsonnet/doc-util/main.libsonnet';
       + route.spec.to.withName(service.metadata.name)
       + route.spec.port.withTargetPort(port),
 
+    '#weightedRouteFor': d.fn(|||
+      weightedRouteFor constructs a openshift route to the specified services. It expects
+      a value for the name, hostname, port, services and defaults to '/' for the path.
+      services expects an array consisting of objects with the keys name (name of the service) and weight.
+    |||, [
+      d.arg('name', d.T.string),
+      d.arg('host', d.T.string, ''),
+      d.arg('port', d.T.number, ''),
+      d.arg('services', d.T.object),
+      d.arg('path', d.T.string, '/'),
+    ]),
+    weightedRouteFor(name, host, port, services, path='/')::
+      route.new(name, host, path)
+      + route.spec.to.withKind('Service')
+      + route.spec.to.withName(services[0].name)
+      + route.spec.to.withWeight(services[0].weight)
+      + route.spec.port.withTargetPort(port)
+      + route.spec.withAlternateBackends(
+        std.prune(
+          std.mapWithIndex(
+            function(i, v) if i == 0 then null else v { kind: 'Service' }, services
```

Comment:
According to the docs, the `kind` field defaults to `Service`. Have you tested leaving this empty?

Reply:
Yes. when leaving empty one gets
`error: error validating "STDIN": error validating data: ValidationError(Route.spec.alternateBackends[0]): missing required field "kind" in com.github.openshift.api.route.v1.RouteTargetReference; if you choose to ignore these errors,
 turn validation off with --validate=false` during tk apply

Reply:
Okay, then let's keep this